### ISSUE TITLE
ReplicaSet pod always in Terminating status in GKE

### ISSUE THREAD

**Issue by noprom** [appscode/voyager#1095]
I'am running `voyager` in GKE. Here is my issue:
### Installing in GKE Cluster
```
gcloud info | grep Account
Account: [user@example.org]
kubectl create clusterrolebinding cluster-admin-binding --clusterrole=cluster-admin --user=user@example.org
```
Then install it:
```
helm repo add appscode https://charts.appscode.com/stable/
helm repo update

curl -fsSL -o onessl https://github.com/kubepack/onessl/releases/download/0.3.0/onessl-linux-amd64 \
  && chmod +x onessl \
  && sudo mv onessl /usr/local/bin/

helm install appscode/voyager --namespace kube-system --name voyager-operator --version 7.0.0 \
  --set cloudProvider=gke \
  --set apiserver.ca="$(onessl get kube-ca)" \
  --set apiserver.enableValidatingWebhook=true
```
Nothing goes wrong when I install it.

### The status of the install:
```
helm status voyager-operator
```
OUTPUT:
```
LAST DEPLOYED: Tue Jun  5 14:19:37 2018
NAMESPACE: kube-system
STATUS: DEPLOYED

RESOURCES:
==> v1/RoleBinding
NAME                                                                       AGE
voyager-voyager-operator-apiserver-extension-server-authentication-reader  17s

==> v1/Service
NAME                      TYPE       CLUSTER-IP     EXTERNAL-IP  PORT(S)                      AGE
voyager-voyager-operator  ClusterIP  10.11.252.223  <none>       443/TCP,56790/TCP,56791/TCP  17s

==> v1/Pod(related)
NAME                                       READY  STATUS   RESTARTS  AGE
voyager-voyager-operator-868b8c76dd-q5r6x  1/1    Running  0         16s

==> v1/ServiceAccount
NAME                      SECRETS  AGE
voyager-voyager-operator  1        17s

==> v1/ClusterRoleBinding
NAME                                               AGE
voyager-voyager-operator-apiserver-auth-delegator  17s
voyager-voyager-operator                           17s

==> v1beta1/Deployment
NAME                      DESIRED  CURRENT  UP-TO-DATE  AVAILABLE  AGE
voyager-voyager-operator  1        1        1           1          17s

==> v1beta1/APIService
NAME                                    AGE
v1beta1.admission.voyager.appscode.com  17s

==> v1beta1/ValidatingWebhookConfiguration
admission.voyager.appscode.com  17s

==> v1/Secret
NAME                                     TYPE    DATA  AGE
voyager-voyager-operator-apiserver-cert  Opaque  2     17s

==> v1/ClusterRole
NAME                      AGE
voyager-voyager-operator  17s


NOTES:
Set cloudProvider for installing Voyager

To verify that Voyager has started, run:

  kubectl --namespace=kube-system get deployments -l "release=voyager-operator, app=voyager"

```
Seems nothing wrong with the install.

### Add an ingress
Then I add an ingress:
```
apiVersion: voyager.appscode.com/v1beta1
kind: Ingress
metadata:
  name: eospace-eosbp-nodeosd
  labels:
    app: eosbp-nodeosd
    chart: eosbp-0.1.1
    release: eospace
    heritage: Tiller
  annotations:
    ingress.appscode.com/type=LoadBalancer
    kubernetes.io/ingress.class=voyager
spec:
  tls:
    - hosts:
        - api.xxx.io
      secretName: xxx-tls-cert
  rules:
    - host: api.xxx.io
      http:
        paths:
          - path: /v1/*
            backend:
              serviceName: eospace-eosbp-nodeosd
              servicePort: 80
    - host: p2p.xxx.io
      tcp:
        port: 88
        backend:
          serviceName: eospace-eosbp-nodeosd
          servicePort: 88
```

And my service:
```
helm status eospace
```
```
LAST DEPLOYED: Tue Jun  5 13:36:21 2018
NAMESPACE: eospace
STATUS: DEPLOYED

RESOURCES:
==> v1beta1/Ingress
NAME                   AGE
eospace-eosbp-nodeosd  2h

==> v1/Pod(related)
NAME                     READY  STATUS   RESTARTS  AGE
eospace-eosbp-bp-0       1/1    Running  0         2h
eospace-eosbp-nodeosd-0  1/1    Running  0         2h

==> v1/Secret
NAME                                            TYPE                     DATA  AGE
eospace-eosbp-gitlab-docker-registry-for-eosbp  kubernetes.io/dockercfg  1     2h

==> v1/ConfigMap
NAME                   DATA  AGE
eospace-eosbp-bp       2     2h
eospace-eosbp-nodeosd  2     2h

==> v1/StorageClass
NAME               PROVISIONER           AGE
gke-vol-expansion  kubernetes.io/gce-pd  2h

==> v1/Service
NAME                   TYPE       CLUSTER-IP   EXTERNAL-IP  PORT(S)        AGE
eospace-eosbp-bp       ClusterIP  10.11.242.8  <none>       80/TCP,88/TCP  2h
eospace-eosbp-nodeosd  ClusterIP  10.11.248.6  <none>       80/TCP,88/TCP  2h

==> v1/StatefulSet
NAME                   DESIRED  CURRENT  AGE
eospace-eosbp-bp       1        1        2h
eospace-eosbp-nodeosd  1        1        2h
```

### Problem
The problem is that I can see a load balancer was created in GKE:
![image](https://user-images.githubusercontent.com/8455444/40982424-ce877036-690f-11e8-91f8-9f50fc7d3da5.png)
![image](https://user-images.githubusercontent.com/8455444/40982557-1547748a-6910-11e8-910f-19488ffb897f.png)

And I can see the logs:
```
kubectl describe pod voyager-eospace-eosbp-nodeosd-75bf898fcf-clkw6 -n eospace
```
```
Name:                      voyager-eospace-eosbp-nodeosd-75bf898fcf-clkw6
Namespace:                 eospace
Node:                      gke-eos-eos-pool-4fd6d449-kl46/10.128.0.11
Start Time:                Tue, 05 Jun 2018 09:33:29 +0000
Labels:                    origin=voyager
                           origin-api-group=voyager.appscode.com
                           origin-name=eospace-eosbp-nodeosd
                           pod-template-hash=3169454979
Annotations:               ingress.appscode.com/last-applied-annotation-keys=
Status:                    Terminating (expires Tue, 05 Jun 2018 09:34:00 +0000)
Termination Grace Period:  30s
IP:
Controlled By:             ReplicaSet/voyager-eospace-eosbp-nodeosd-75bf898fcf
Containers:
  haproxy:
    Container ID:
    Image:         appscode/haproxy:1.8.8-7.0.0-alpine
    Image ID:
    Ports:         80/TCP, 88/TCP
    Args:
      --enable-analytics=true
      --burst=1000000
      --cloud-provider=gke
      --ingress-api-version=voyager.appscode.com/v1beta1
      --ingress-name=eospace-eosbp-nodeosd
      --qps=1e+06
      --logtostderr=false
      --alsologtostderr=false
      --v=3
      --stderrthreshold=0
    State:          Terminated
      Exit Code:    0
      Started:      Mon, 01 Jan 0001 00:00:00 +0000
      Finished:     Mon, 01 Jan 0001 00:00:00 +0000
    Ready:          False
    Restart Count:  0
    Environment:
      APPSCODE_ANALYTICS_CLIENT_ID:  6e5bb4324265c7d311c1081f06e22e06
    Mounts:
      /etc/ssl/private/haproxy from voyager-certdir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from voyager-eospace-eosbp-nodeosd-token-xdv8t (ro)
Conditions:
  Type           Status
  Initialized    True
  Ready          False
  PodScheduled   True
Volumes:
  voyager-certdir:
    Type:    EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
  voyager-eospace-eosbp-nodeosd-token-xdv8t:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  voyager-eospace-eosbp-nodeosd-token-xdv8t
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:          <none>
```
This pod is always `Terminating` and not even have any events.
What's wrong with that?

### Version
```
kubectl version
```
```
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.7", GitCommit:"dd5e1a2978fd0b97d9b78e1564398aeea7e7fe92", GitTreeState:"clean", BuildDate:"2018-04-19T00:05:56Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"10+", GitVersion:"v1.10.2-gke.3", GitCommit:"d2c7a2bd41036f9474287579a725dc54c904e92d", GitTreeState:"clean", BuildDate:"2018-05-23T00:19:39Z", GoVersion:"go1.9.3b4", Compiler:"gc", Platform:"linux/amd64"}
```
```
helm version
```
```
Client: &version.Version{SemVer:"v2.8.1", GitCommit:"6af75a8fd72e2aa18a2b278cfe5c7a1c5feca7f2", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.8.1", GitCommit:"6af75a8fd72e2aa18a2b278cfe5c7a1c5feca7f2", GitTreeState:"clean"}
```

Any advice would be appreciated! @tamalsaha

**Comment by tamalsaha** [comment]
@noprom , fyi, there is a Kernel bug that affects GKE 1.10 cluster that uses COS image. If you use Ubuntu , then things are ok. https://github.com/appscode/voyager/issues/1084#issuecomment-393923468

Can you find the node where this pod was running, then run `docker ps -a` which should show the container id, then `docker logs <container-id>` to find the log?

**Comment by noprom** [comment]
Hi @tamalsaha , thanks for the reply.
Actually this pod doesn't even create a container for me because when I run:
```
kubectl describe pod voyager-eospace-eosbp-nodeosd-75bf898fcf-clkw6 -n eospace
```
The result:
```
Name:                      voyager-eospace-eosbp-nodeosd-75bf898fcf-clkw6
Namespace:                 eospace
Node:                      gke-eos-eos-pool-4fd6d449-kl46/10.128.0.11
Start Time:                Tue, 05 Jun 2018 09:33:29 +0000
Labels:                    origin=voyager
                           origin-api-group=voyager.appscode.com
                           origin-name=eospace-eosbp-nodeosd
                           pod-template-hash=3169454979
Annotations:               ingress.appscode.com/last-applied-annotation-keys=
Status:                    Terminating (expires Tue, 05 Jun 2018 09:34:00 +0000)
Termination Grace Period:  30s
IP:
Controlled By:             ReplicaSet/voyager-eospace-eosbp-nodeosd-75bf898fcf
Containers:
  haproxy:
    Container ID:
    Image:         appscode/haproxy:1.8.8-7.0.0-alpine
    Image ID:
    Ports:         80/TCP, 88/TCP
    Args:
      --enable-analytics=true
      --burst=1000000
      --cloud-provider=gke
      --ingress-api-version=voyager.appscode.com/v1beta1
      --ingress-name=eospace-eosbp-nodeosd
      --qps=1e+06
      --logtostderr=false
      --alsologtostderr=false
      --v=3
      --stderrthreshold=0
    State:          Terminated
      Exit Code:    0
      Started:      Mon, 01 Jan 0001 00:00:00 +0000
      Finished:     Mon, 01 Jan 0001 00:00:00 +0000
    Ready:          False
    Restart Count:  0
    Environment:
      APPSCODE_ANALYTICS_CLIENT_ID:  6e5bb4324265c7d311c1081f06e22e06
    Mounts:
      /etc/ssl/private/haproxy from voyager-certdir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from voyager-eospace-eosbp-nodeosd-token-xdv8t (ro)
Conditions:
  Type           Status
  Initialized    True
  Ready          False
  PodScheduled   True
Volumes:
  voyager-certdir:
    Type:    EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
  voyager-eospace-eosbp-nodeosd-token-xdv8t:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  voyager-eospace-eosbp-nodeosd-token-xdv8t
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:          <none>
```
It shows that the `    Container ID:` part is missing.
So does this mean that it doesn't even create a container for me?

**Comment by tamalsaha** [comment]
`kubectl describe deploy voyager-eospace-eosbp-nodeosd -n eospace` ? If you come to our https://slack.appscode.com/ and ping me (`@tamal`), it will be easy to debug. I think there is some mistake in the yaml.

**Comment by noprom** [comment]
Hi @tamalsaha , here is my result:
```
Name:                   voyager-eospace-eosbp-nodeosd
Namespace:              eospace
CreationTimestamp:      Tue, 05 Jun 2018 09:04:00 +0000
Labels:                 app=eosbp-nodeosd
                        chart=eosbp-0.1.1
                        heritage=Tiller
                        origin=voyager
                        origin-api-group=voyager.appscode.com
                        origin-name=eospace-eosbp-nodeosd
                        release=eospace
Annotations:            deployment.kubernetes.io/revision=1
                        ingress.appscode.com/origin-api-schema=voyager.appscode.com/v1beta1
                        ingress.appscode.com/origin-name=eospace-eosbp-nodeosd
Selector:               origin=voyager,origin-api-group=voyager.appscode.com,origin-name=eospace-eosbp-nodeosd
Replicas:               2 desired | 0 updated | 0 total | 0 available | 1 unavailable
StrategyType:           RollingUpdate
MinReadySeconds:        0
RollingUpdateStrategy:  25% max unavailable, 25% max surge
Pod Template:
  Labels:           origin=voyager
                    origin-api-group=voyager.appscode.com
                    origin-name=eospace-eosbp-nodeosd
  Annotations:      ingress.appscode.com/last-applied-annotation-keys=
  Service Account:  voyager-eospace-eosbp-nodeosd
  Containers:
   haproxy:
    Image:  appscode/haproxy:1.8.8-7.0.0-alpine
    Ports:  80/TCP, 88/TCP, 443/TCP
    Args:
      --enable-analytics=true
      --burst=1000000
      --cloud-provider=gke
      --ingress-api-version=voyager.appscode.com/v1beta1
      --ingress-name=eospace-eosbp-nodeosd
      --qps=1e+06
      --logtostderr=false
      --alsologtostderr=false
      --v=3
      --stderrthreshold=0
    Environment:
      APPSCODE_ANALYTICS_CLIENT_ID:  6e5bb4324265c7d311c1081f06e22e06
    Mounts:
      /etc/ssl/private/haproxy from voyager-certdir (rw)
  Volumes:
   voyager-certdir:
    Type:    EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
Conditions:
  Type           Status  Reason
  ----           ------  ------
  Progressing    True    NewReplicaSetAvailable
  Available      False   MinimumReplicasUnavailable
OldReplicaSets:  voyager-eospace-eosbp-nodeosd-75bf898fcf (0/1 replicas created)
NewReplicaSet:   <none>
Events:          <none>
```

**Comment by tamalsaha** [comment]
Fixed in 1.9.6
